From 15aad47d22da665b056b4698da1ced1c914dfbad Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:41 -0400
Subject: [tools] perf/lock: Return proper code in report_lock_*_event

Message-id: <1408462094-14194-113-git-send-email-jolsa@redhat.com>
Patchwork-id: 88111
O-Subject: [PATCH RHEL7.1 BZ1131394 112/325] perf lock: Return proper code in report_lock_*_event
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit b33492ade49a223a666e582d0c63566609e7014b
Author: Davidlohr Bueso <davidlohr@hp.com>
Date: Sun Sep 8 19:19:14 2013 -0700

description
===========
The report_lock_*_event() functions return -1 when lock_stat_findnew(),
thread_stat_findnew() or get_seq() return NULL. These functions only
return this value when failing to allocate memory, this return -ENOMEM
instead.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-lock.c b/tools/perf/builtin-lock.c
index 148f7e2..d318862 100644
--- a/tools/perf/builtin-lock.c
+++ b/tools/perf/builtin-lock.c
@@ -400,17 +400,17 @@ static int report_lock_acquire_event(struct perf_evsel *evsel,
 
  ls = lock_stat_findnew(addr, name);
  if (!ls)
-  return -1;
+  return -ENOMEM;
  if (ls->discard)
   return 0;
 
  ts = thread_stat_findnew(sample->tid);
  if (!ts)
-  return -1;
+  return -ENOMEM;
 
  seq = get_seq(ts, addr);
  if (!seq)
-  return -1;
+  return -ENOMEM;
 
  switch (seq->state) {
  case SEQ_STATE_UNINITIALIZED:
@@ -472,17 +472,17 @@ static int report_lock_acquired_event(struct perf_evsel *evsel,
 
  ls = lock_stat_findnew(addr, name);
  if (!ls)
-  return -1;
+  return -ENOMEM;
  if (ls->discard)
   return 0;
 
  ts = thread_stat_findnew(sample->tid);
  if (!ts)
-  return -1;
+  return -ENOMEM;
 
  seq = get_seq(ts, addr);
  if (!seq)
-  return -1;
+  return -ENOMEM;
 
  switch (seq->state) {
  case SEQ_STATE_UNINITIALIZED:
@@ -533,17 +533,17 @@ static int report_lock_contended_event(struct perf_evsel *evsel,
 
  ls = lock_stat_findnew(addr, name);
  if (!ls)
-  return -1;
+  return -ENOMEM;
  if (ls->discard)
   return 0;
 
  ts = thread_stat_findnew(sample->tid);
  if (!ts)
-  return -1;
+  return -ENOMEM;
 
  seq = get_seq(ts, addr);
  if (!seq)
-  return -1;
+  return -ENOMEM;
 
  switch (seq->state) {
  case SEQ_STATE_UNINITIALIZED:
@@ -587,17 +587,17 @@ static int report_lock_release_event(struct perf_evsel *evsel,
 
  ls = lock_stat_findnew(addr, name);
  if (!ls)
-  return -1;
+  return -ENOMEM;
  if (ls->discard)
   return 0;
 
  ts = thread_stat_findnew(sample->tid);
  if (!ts)
-  return -1;
+  return -ENOMEM;
 
  seq = get_seq(ts, addr);
  if (!seq)
-  return -1;
+  return -ENOMEM;
 
  switch (seq->state) {
  case SEQ_STATE_UNINITIALIZED:
-- 
1.7.1